google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiManager.java

```diff
@@ -137,6 +140,28 @@ private static void enableApi(
         .execute();
   }
 
+  /**
+   * Fetches the list of {@link Role} for the supplied {@link CloudProject} by querying the Iam API.
+   */
+  static List<Role> getServiceAccountRoles(CloudProject cloudProject) {
```

Comment:
What do you think about using valid `CredentialedUser` as a parameter for this method? It seems we are getting list of all roles for the full account rather than for a specific project so it may make this method a bit cleaner.

Reply:
I'd like to isolate the Credential management to the API class itself. That being said, some refactoring I'd like to do in a different PR is to make this class non-static and do the credential management only once.